src/commands/remove.rs

```diff
@@ -1,43 +1,59 @@
 use serenity::{
-    client::Context,
-    framework::standard::{macros::command, Args, CommandResult},
-    model::channel::Message,
+    client::Context, model::interactions::application_command::ApplicationCommandInteraction,
+    prelude::SerenityError,
 };
 
 use crate::{
-    strings::{MISSING_INDEX_QUEUE, NO_SONG_ON_INDEX, NO_VOICE_CONNECTION, QUEUE_IS_EMPTY},
-    utils::send_simple_message,
+    strings::{NO_SONG_ON_INDEX, NO_VOICE_CONNECTION, QUEUE_IS_EMPTY},
+    utils::create_response,
 };
 
-#[command]
-#[aliases("rm")]
-async fn remove(ctx: &Context, msg: &Message, mut args: Args) -> CommandResult {
-    let guild_id = msg.guild(&ctx.cache).await.unwrap().id;
+pub async fn remove(
+    ctx: &Context,
+    interaction: &mut ApplicationCommandInteraction,
+) -> Result<(), SerenityError> {
+    let guild_id = interaction.guild_id.unwrap();
     let manager = songbird::get(ctx).await.unwrap();
 
     let call = match manager.get(guild_id) {
         Some(call) => call,
-        None => return send_simple_message(&ctx.http, msg, NO_VOICE_CONNECTION).await,
+        None => return create_response(&ctx.http, interaction, NO_VOICE_CONNECTION).await,
     };
 
-    let remove_index: usize = match args.single::<usize>() {
-        Ok(t) => t,
-        Err(_) => return send_simple_message(&ctx.http, msg, MISSING_INDEX_QUEUE).await,
-    };
+    let args = interaction.data.options.clone();
+
+    let remove_index = args
+        .first()
+        .unwrap()
+        .value
+        .as_ref()
+        .unwrap()
+        .as_u64()
+        .unwrap() as usize;
 
     let handler = call.lock().await;
     let queue = handler.queue().current_queue();
 
     if queue.is_empty() {
-        send_simple_message(&ctx.http, msg, QUEUE_IS_EMPTY).await
+        create_response(&ctx.http, interaction, QUEUE_IS_EMPTY).await
     } else if queue.len() < remove_index + 1 {
-        send_simple_message(&ctx.http, msg, NO_SONG_ON_INDEX).await
+        create_response(&ctx.http, interaction, NO_SONG_ON_INDEX).await
     } else if remove_index == 0 {
-        send_simple_message(&ctx.http, msg, "Can't remove currently playing song!").await
+        create_response(
+            &ctx.http,
+            interaction,
+            "Can't remove currently playing song!",
+        )
+        .await
```

Comment:
What do you think about just not doing anything?

Reply:
Nice that you point it out. I would even go further and say, if the index is 0, just call the `skip` command that we already have!

Reply:
Personally, I think the user that executes `/remove 0` is more likely to have made a mistake with indexes than actually trying to skip the first song. There's a `/skip` command for that.

Regardless, if this change were to happen, it should be on a different branch.